libs/dh/wholesale/feature-calculation-steps/src/lib/actors/dh-wholesale-actors.component.html

```diff
@@ -16,27 +16,27 @@
 -->
 <ng-container
   *rxLet="
-    energySuppliersForConsumption$;
-    let energySuppliers;
+    actors$;
+    let actors;
     suspense: loading;
     error: error;
     errorTrigger: errorTrigger$
   "
 >
-  <ng-container *transloco="let t; read: 'wholesale.energySuppliers'">
+  <ng-container *transloco="let t; read: 'wholesale.actors'">
     <h5 class="watt-on-light--high-emphasis">
-      {{ t("amountOfResults", { length: (energySuppliers || []).length }) }}
+      {{ t("amountOfResults", { length: (actors || []).length }) }}
     </h5>
     <watt-table
       [resolveHeader]="t"
-      [activeRow]="getSelectedEnergySupplier(energySuppliers)"
-      [activeRowComparator]="isSelectedEnergySupplier"
-      [dataSource]="_dataSource(energySuppliers)"
+      [activeRow]="getSelectedActor(actors)"
+      [activeRowComparator]="isSelectedActor"
+      [dataSource]="actors | wattDataSource: _dataSource"
       [columns]="_columns"
       (rowClick)="rowClick.emit($event)"
     >
     </watt-table>
-    <watt-paginator></watt-paginator>
+    <watt-paginator (initialized)="setPaginator($event)"></watt-paginator>
```

Comment:
Question: This `initialized` event seems very specific to this use case (rendering paginator async), so I fear it might be a little confusing for other devs whether they should always use this event or not. Have you considered using the `changes` property with `ViewChildren` instead?

I do like the convenience here though, so alternatively to make it even more convenient, would it be possible to do something like this (simplified):

```html
<watt-table #actors [dataSource]="_dataSource" [columns]="_columns"></watt-table>
<watt-paginator [for]="actors"></watt-paginator>
```

Then in the `watt-paginator` the `for` input would be a setter that sets the paginator instance on the tables data source. Could be pretty cool if (at all possible) and would eliminate the need for any logic in the component itself 🙂

Reply:
Agreed, that would be an excellent solution! I will try to implement that.